packages/marshal/src/helpers/passStyleHelpers.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
+
+import '../types.js';
+import './internal-types.js';
+/**
+ * TODO Why do I need these?
+ *
+ * @typedef {import('./internal-types.js').Checker} Checker
+ */
+import '@agoric/assert/exported.js';
+
+const { details: X, quote: q } = assert;
+const { hasOwnProperty: objectHasOwnProperty } = Object;
+const { apply } = Reflect;
+const { isArray } = Array;
+
+export const hasOwnPropertyOf = (obj, prop) =>
+  apply(objectHasOwnProperty, obj, [prop]);
+harden(hasOwnPropertyOf);
+
+export const isPrimitive = val => Object(val) !== val;
+harden(isPrimitive);
+
+export const PASS_STYLE = Symbol.for('passStyle');
+
+/**
+ * For a function to be a valid method, it must not be passable.
+ * Otherwise, we risk confusing pass-by-copy data carrying
+ * far functions with attempts at far objects with methods.
+ *
+ * TODO HAZARD Because we check this on the way to hardening a remotable,
+ * we cannot yet check that `func` is hardened. However, without
+ * doing so, it's inheritance might change after the `PASS_STYLE`
+ * check below.
+ *
+ * @param {*} func
```

Comment:
As per Agoric conventions, prefer `any` over `*` to indicate that you considered what the type should be and it should really be the top type, instead of just being too lazy to type it.

```suggestion
 * @param {any} func
```